web_dwc2.py

```diff
@@ -623,467 +613,251 @@ def rr_reply(self, web_):
-            "firmwareName": "Klipper",
-            "geometry": self.kin_name,
-            "axes": len(self.get_axes_homed()),
-            "totalAxes": len(self.get_axes_homed()) + len([1 for ex_ in extr_stat]),
-            "axisNames": "XYZ",  # + "".join([ "U" for ex_ in extr_stat ]),
-            "volumes": 1,
-            "mountedVolumes": 1,
-            "name": self.printername,
-            "probe": {
-                "threshold": 100,
-                "height": 0,
-                "type": 8
-            },
-            "tools": [
-                {
-                    "number": extr_stat.index(ex_),
-                    "name": ex_['name'],
-                    "heaters": [extr_stat.index(ex_) + 1],
-                    "drives": [extr_stat.index(ex_)],
-                    "axisMap": [1],
-                    "fans": 1,
-                    "filament": "",
-                    "offsets": [0, 0, 0]
-                } for ex_ in extr_stat]
-            ,
-            # "vin": {
-            #	"min": 24.2,
-            #	"cur": 24.3,
-            #	"max": 24.5
-            # }
-            # https://github.com/chrishamm/DuetWebControl/blob/92daf5d98db5def091a3f1a9d4324e1b953fc649/src/store/machine/connector/PollConnector.js#L470
-            # "output": {
-            #	"message": "Titten!"
-            # }
-            # "output": {
-            #	"msgBox": {
-            #		"mode": 2,
-            #		"title": "testtitle",
-            #		"msg": "hello world",
-            #		"timeout": 5000,	#	can be zero?
-            #		"controls": 1
-            #	}
-            # }
-        })
-
-        if self.message:
-            self.status_2.update(
-                self.message
-            )
-
-        if self.popup:
-            self.status_2.update(
-                self.popup
-            )
+                    "current": heater_bed_status['temperature'],
+                    "active": heater_bed_status['target'],
+                    "state": state,
+                    "heater": temp_numbers
+                }
+            })
 
-        if self.chamber:
-            chamber_stats = self.chamber.get_status(0)
-            self.status_2['temps'].update({
+            temp_currents = temp_currents + [heater_bed_status['temperature']]
+            temp_states = temp_states + [state]
+            temp_names = temp_names + ["Heater " + str(temp_numbers)]
+            temp_numbers = temp_numbers + 1
+
+        ex_min_extrude_temp = 0
+        ex_max_temp = 0
+        if self.extruders is not None:
+            for ex_ in self.extruders:
+                if ex_ is not None:
+                    ex_status = ex_.heater.get_status(now)
+                    state = 0
+                    if (ex_status['target'] > 0):
+                        state = 2
+
+                    if (ex_min_extrude_temp > ex_.heater.min_extrude_temp):
+                        ex_min_extrude_temp =  ex_.heater.min_extrude_temp
```

Comment:
There is a usersetting in klipper fot min extrude temp. We should honor that instead here.

Reply:
then you can use max() in one line.